capi_problems.rst

```diff
@@ -70,7 +70,141 @@ and the particular requirements that each of them has.
 C API Stakeholders
 ==================
 
-[TODO: write this section]
+As mentioned in the introduction, the C API was originally
+created as the internal interface between CPython's
+interpreter and the Python layer. It was later exposed as
+a way for third party developers to extend and embed Python
+programs. Over the years, new types of stakeholders emerged,
+with different requirements and areas of focus. This section
+describes this complex state of affairs in terms of the
+actions that different stakeholders need to perform through
+the C API.
+
+There are actions which are generic, and required by
+all types of API users:
+
+* Define functions and call them
+* Create classes and instantiate them
+* Create instances of builtin and user defined types
+  and perform operations on them
+* Introspect objects, including types, instances, and functions.
+* Raise and handle exceptions
+* Import modules
+* Manage threads
+* Manage sub-interpreters
+* Handle and send signals
+
+Groups of users are:
+
+**Extension writers**
+
+These are the traditional users of the C API, and their requirements
+are listed above.
+
+**Embedders**
+
+Applications with an embedded Python interpreter. Example are
+`Blender <https://docs.blender.org/api/current/info_overview.html>`__ and
+`OBS <https://obsproject.com/wiki/Getting-Started-With-OBS-Scripting>`__.
+
+They need to be able to
+
+* Configure the interpreter (import paths, inittab, sys.argv, ...)
+* Interact with the execution model and program lifetime, including
+  clean interpreter shutdown and restart
+* Represent complex data models in a way Python can use without
+  having to create deep copies.
+* Frozen modules.
+* Run multiple independent interpreters (in particular, when embedded
+  in a library that wants to avoid global effects).
+
+**Alternative Python Implementations**
+
+Alterantive implementations of Python (such as
+`PyPy <https://www.pypy.org>`__,
+`GraalPython <https://www.graalvm.org/python/>`__,
+and `Jython <https://www.jython.org>`__), may take
+very different approaches for the implementation of
+different subsystems. They need:
+
+* The API to be abstract and hide implementation details.
+* A specification of the API, ideally with a test suite
+  that ensures compatibility.
+* An ABI that can be shared across Python implementations.
```

Comment:
Portable ABI is nice, but I don't think that it's a "need". Maybe just express it differently.